lms/templates/student_account/register.underscore

```diff
@@ -29,11 +29,17 @@
             <%
             _.each( context.providers, function( provider) {
                 if ( provider.registerUrl ) { %>
-                    <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
-                        <span class="icon fa <%- provider.iconClass %>" aria-hidden="true"></span>
-                        <span aria-hidden="true"><%- provider.name %></span>
-                        <span class="sr"><%- _.sprintf( gettext("Create account using %(providerName)s."), {providerName: provider.name} ) %></span>
-                    </button>
+                    <div class="button-container" <% if ( provider.buttonColor ) { %>style="color: <%- provider.buttonColor %>; background-color: <%- provider.buttonColor %>"<% } %>>
+                        <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
+                            <div class="icon <% if ( provider.iconClass ) { %>fa <%- provider.iconClass %><% } %>" <% if ( provider.buttonColor ) { %>style="background: <%- provider.buttonColor %>"<% } %> aria-hidden="true">
+                                <% if ( provider.iconImage ) { %>
+                                    <img src="<%- provider.iconImage %>" alt="<%- provider.name %> icon" />
+                                <% } %>
+                            </div>
+                            <span <% if ( provider.textColor ) { %>style="color: <%- provider.textColor %>"<% } %> aria-hidden="true"><%- provider.name %></span>
```

Comment:
@AlasdairSwan @douglashall @omarkhan 
I need to de-risk this ASAP so that we can make this Friday's release. Our main focus for this project is adding the AzureAD provider, but we want to do so in a generic way to avoid introducing tech debt. So: Are there any objections to this plan?
* we add classes for the AzureAD provider to the global sass files where the styles for Google, Facebook, and LinkedIn already reside
* We remove "background color" and "text color" from the django admin
* We remove the inline styles from this PR, which would no longer be needed

With the above changes, we wouldn't provide an explicit mechanism for styling other custom providers, but it could be done by adding custom sass to any instance, either through a theme or some other mechanism.